src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -0,0 +1,162 @@
+/-
+Copyleft 2020 Johan Commelin. No rights reserved.
+Authors: Johan Commelin
+-/
+
+import data.mv_polynomial
+import data.fintype.card
+
+/-!
+# Homogeneous polynomials
+
+A multivariate polynomial `φ` is homogeneous of degree `n`
```

Comment:
Can you expand with more of the information in the doc guidelines?

Reply:
I wrote a bit more.